### PR TITLE
2.1.0 release news. [skip ci]

### PR DESCRIPTION
**Pull request by trivialfis**

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -3,6 +3,183 @@ XGBoost Change Log
...
+- Supports thread-based workers.
+- Improved handling for worker errors, including better error message when one of the peers dies during training.
+- Work with IPv6. Currently, this is only supported by the dask interface.
+- Builtin support for various operations like broadcast, allgatherV, allreduce, etc.
+
+Related PRs (#9597, #9576, #9523, #9524, #9593, #9596, #9661, #10319, #10152, #10125, #10332, #10306, #10208, #10203, #10199, #9784, #9777, #9773, #9772, #9759, #9745, #9695, #9738, #9732, #9726, #9688, #9681, #9679, #9659, #9650, #9644, #9649, #9917, #9990, #10313, #10315, #10112, #9531, #10075, #9805, #10198).
+
+The existing option of using `MPI` in RABIT is removed in the release. (#9525)
+
+### NCCL is no-longer bundled in XGBoost for PyPI packages.
```
Comment (trivialfis) [inline NEWS.md:26]: NCCL is now fetched from pypi

**Comment by trivialfis** [comment]
cc @hcho3 @wbo4958 @rongou  @razdoburdin @RAMitchell

File: NEWS.md [hunk 2]
```diff
@@ -3,6 +3,183 @@ XGBoost Change Log
...
+
+### Federated Learning
+Progress has been made on federated learning with improved support for column-split, including the following updates:
+- Column split work for both CPU and GPU. In addition, categorical data is now compatible with column split. (#9562, #9609, #9611, #9628, #9539, #9578, #9685, #9623, #9613, #9511, #9384, #9595)
+-  The use of UBJson to serialize split entries for column split has been implemented, aiding vector-leaf with column-based data split. (#10059, #10055, #9702)
+- Documentation and small fixes. (#9610, #9552, #9614, #9867)
+
+### Ongoing work for SYCL support.
+
+XGBoost is developing a SYCL plugin for SYCL devices, starting with the `hist` tree method. (#10216, #9800, #10311, #9691, #10269, #10251, #10222, #10174, #10080, #10057, #10011, #10138, #10119, #10045, #9876, #9846, #9682)
```
Comment (razdoburdin) [inline NEWS.md:47]: XGBoost is developing a SYCL plugin for SYCL devices, starting with the inference (#9800) and the `hist` tree method. (#10216, #10311, #9691, #10269, #10251, #10222, #10174, #10080, #10057, #10011, #10138, #10119, #10045, #9876, #9846, #9682)

Comment (razdoburdin) [reply NEWS.md:47]: From a release note perspective, we can mention that XGBoost now supports launchable inference on SYCL devices, and that work on adding SYCL support for training is ongoing.

Looking ahead, I plan to complete the upstreaming of training within the next few months and then focus on improving test coverage for SYCL, particularly for Python tests.

File: NEWS.md [hunk 3]
```diff
@@ -3,6 +3,183 @@ XGBoost Change Log
...
+- Work with IPv6. Currently, this is only supported by the dask interface.
+- Built-in support for various operations like broadcast, allgatherV, allreduce, etc.
+
+Related PRs (#9597, #9576, #9523, #9524, #9593, #9596, #9661, #10319, #10152, #10125, #10332, #10306, #10208, #10203, #10199, #9784, #9777, #9773, #9772, #9759, #9745, #9695, #9738, #9732, #9726, #9688, #9681, #9679, #9659, #9650, #9644, #9649, #9917, #9990, #10313, #10315, #10112, #9531, #10075, #9805, #10198).
+
+The existing option of using `MPI` in RABIT is removed in the release. (#9525)
+
+### NCCL is now fetched from PyPI.
+In the previous version, XGBoost statically linked NCCL, which significantly increased the binary size and led to hitting the PyPI repository limit. However, with the new release, we have made a significant improvement. The new release can now dynamically load NCCL from an external source, reducing the binary size. For the PyPI package, the `nvidia-nccl-cu12` package will be fetched during installation. With more downstream packages reusing NCCL, we expect the user environments to be slimmer in the future as well. (#9796, #9804)
```
Comment (rongou) [inline NEWS.md:27]: Probably don't need the "however" here.

**Review by rongou — APPROVED** [review]

File: NEWS.md [hunk 4]
```diff
@@ -3,6 +3,183 @@ XGBoost Change Log
...
+- Small optimization to external memory with a thread pool. This reduces the number of threads launched during iteration. (#9605, #10288)
+
+### Deprecation and breaking changes
+Package-specific breaking changes are outlined in respective sections. Here we list general breaking changes in this release:
+- The command line interface is deprecated due to the increasing complexity of the machine learning ecosystem. Building a machine learning model using a command shell is no longer feasible and could mislead newcomers. (#9485)
+- `Universal binary JSON` is now the default format for saving models (#9947, #9958, #9954, #9955). See https://github.com/dmlc/xgboost/issues/7547 for more info.
+- The `XGBoosterGetModelRaw` is now removed after deprecation in 1.6. (#9617)
+- Drop support for loading remote files. This feature lacks any test. Users are encouraged to use dedicated libraries to fetch remote content. (#9504)
+- Remove the dense libsvm parser plugin. This plugin is never tested or documented (#9799)
+- Make the columnar data setter more general (#9905)
```
Comment (trivialfis) [inline NEWS.md:61]: This is not a breaking change.

**Comment by trivialfis** [comment]
Close https://github.com/dmlc/xgboost/issues/10402

I won't move the original one.

**Review by hcho3 — APPROVED** [review]

**Pull request merged**